### PR TITLE
Fix invalid null constraint and FQN

### LINKED ISSUE #516: [BUG]: Identified bugs in V 0.3.0
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Category of Bug / Issue

ReconcileError

### Current Behavior

1. Fix DDL - remove not null constraint from main table DDL
2. connector/datbricks.py -- file for line 52 instead of `table_with_namespace = f"{catalog}.{schema}.{table}"`  should be replaced with `table_with_namespace = f"{catalog}.{schema}.{table}" if catalog else f"{schema}.{table}"`  To handle source_catalog as null

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Relevant log output or Exception details

_No response_

### Sample Query

_No response_

### Operating System

macOS

### Version

latest via Databricks CLI

### PR DESCRIPTION
**Pull request by bishwajit-db**
Fixes #516

### CODE REVIEW

**Review by sundarshankar89 — APPROVED** [review]
LGTM

**Review by ganeshdogiparthi-db — APPROVED** [review]
LGTM

**Comment by ganeshdogiparthi-db** [comment]
Please hold on. Dont merge.

**Comment by ganeshdogiparthi-db** [comment]
Pushed a small change in the DDL. Tested on this branch by installing and creating jar and using it in workflow.

File: src/databricks/labs/remorph/resources/queries/reconcile/installation/main.sql [hunk 1]
```diff
@@ -3,10 +3,10 @@ CREATE TABLE IF NOT EXISTS main (
     recon_id STRING NOT NULL,
     source_type STRING NOT NULL,
     source_table STRUCT<
-                         catalog: STRING NOT NULL,
+                         catalog: STRING,
                          schema: STRING NOT NULL,
                          table_name: STRING NOT NULL
-                        > NOT NULL,
```
Comment (vijaypavann-db) [inline src/databricks/labs/remorph/resources/queries/reconcile/installation/main.sql:9]: Why `NOT NULL` constraint is removed for `source_table`?
For `reconciliation` we expect both `source_table` and `target_table` details, isn't it?

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**